packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -0,0 +1,95 @@
+import * as cdk from '@aws-cdk/core';
+import { InputValidator } from './private/validation';
+import { CfnApplication } from './servicecatalogappregistry.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * A Service Catalog AppRegistry Application.
+ */
+export interface IApplication extends cdk.IResource {
+  /**
+   * The ARN of the application.
+   * @attribute
+   */
+  readonly applicationArn: string;
+
+  /**
+   * The ID of the application.
+   * @attribute
+   */
+  readonly applicationId: string;
+}
+
+/**
+ * Properties for a Service Catalog AppRegistry Application
+ */
+export interface ApplicationProps {
+  /**
+   * Enforces a particular physical application name.
+   */
+  readonly applicationName: string;
+
+  /**
+   * Description for application.
+   * @default - No description provided
+   */
+  readonly description?: string;
+}
+
+abstract class ApplicationBase extends cdk.Resource implements IApplication {
+  public abstract readonly applicationArn: string;
+  public abstract readonly applicationId: string;
+}
+
+/**
+ * A Service Catalog AppRegistry Application.
+ */
+export class Application extends ApplicationBase implements IApplication {
```

Comment:
No need for this `implements IApplication` here:

```suggestion
export class Application extends ApplicationBase {
```